### DISCUSSION TITLE
[Bug]: Unable to combine Pimcore X with HWIOauthBundle for OAuth login

### DISCUSSION THREAD

**Discussion by schaetzle** [pimcore/pimcore#11307] [Q&A / Support] [answered]
### Expected behavior

We use the HWIOauthBundle for connecting to Keycloak in one of our Pimcore 6 projects. 

Behavior during login is as follows: You click on login, you are redirected to Keycloak, there you log into your account, you come back to our app and you are authenticated as a frontend user. Also you are not logged out from Pimcore admin. 

This is also the expected Behavior for our fresh Pimcore X project with HWIOauthBundle.

### Actual behavior

The first login (when you have to login into your Keycloak account) is not registered correctly. 
You come back from Keycloak, but you are not authenticated in our frontend firewall. 
Also you are automatically logged out from pimcore admin. 

A second login try afterwards however is successful. You click on login, you are redirected to Keycloak, Keycloak recognizes you and redirects you back to our app. There you are authenticated as a frontend user. And also you are still logged in into pimcore admin.

So the first login process (when you first have to login to Keycloak) is not working correctly.

### Steps to reproduce

We setup a fresh Pimcore X installation and installed the HWIOauthBundle according to the documentation. We tried dev-master and also version 1.3. 

We also tried to install the way that is described in the [Pimcore Customer Management Framework documentation](https://pimcore.com/docs/customer-management-framework/current/Single_Sign_On.html). But this seems to be outdated. Many errors during installation and setup process. And to be honest we don't want to install the whole CMF just for having a Single Sign On.

It could be that the Pimcore firewall and our frontend firewall get in each other's way. But we have tested many different configurations. It just seems not to be possible to use both of them.

Does anyone of you have a similar case and got Pimcore X working with HWIOauthBundle?

Greetings
Chris

**Comment by schaetzle** [comment, marked as answer]
Hi, 

after implementing our own Keycloak connection, we found the problem.
It's the cookie_samesite setting in symfony framework config: https://symfony.com/doc/5.4/reference/configuration/framework.html#cookie-samesite

Pimcore X uses "strict". We simply changed that setting to the Symfony default value "lax" and all oauth connections work fine.

So this ticket can be closed.

Greetings
Chris

**Reply by fashxp** [reply]
thx for sharing your insights!
cookie_samesite setting was changed due to security reasons, details see also here: https://github.com/pimcore/pimcore/pull/9885
